### DISCUSSION TITLE
reducing build time

### DISCUSSION THREAD

**Discussion by MikeMazz23** [darkreader/darkreader#10804] [Q&A] [answered]
Is there any way to reduce the build time? It easily takes over 5 minutes each time I run the build. The build-js task is what takes all of the time. There are no errors. Is there any way to reduce that? This makes it very inefficient when trying to make updates for the project.

Does it take several minutes for anyone else?

**Comment by bershanskiy** [comment]
Hi, @MikeMazz23 by default, Dark Reader builds for all platforms and that does take a lot of time. However, we support build flags which reduce the number of files built and speed up the build. Specifically, consider using one of `--api`, `--chrome`, `--chrome-mv3`, `--firefox`, or `--thunderbird`. Also, you might want to use incremental builds with `--watch`. If you could explain the scope of your changes, I might give more specific advice.

**Comment by bershanskiy** [comment]
Related: #10806.

**Comment by bershanskiy** [comment, marked as answer]
Please fetch the latest code and run `npm run build -- --help` to see information about build flags.
